### ISSUE TITLE
The returned adv images of DeepFool and CWL2 are not correct.

### ISSUE THREAD

**Issue by hyhchaos** [bethgelab/foolbox#554]
Hello, when I use DeepFool (LinfDeepFoolAttack()) to produce adversarial images, I find that the returned adversarial images can not achieve the accuracy returned by the function "attack".

I write some simple code by combining the examples of the tutorial to show this.

```
#!/usr/bin/env python3
import torchvision.models as models
import eagerpy as ep
from foolbox import PyTorchModel, accuracy, samples
import foolbox.attacks as fa
import numpy as np


if __name__ == "__main__":
    # instantiate a model
    model = models.resnet18(pretrained=True).eval()
    preprocessing = dict(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], axis=-3)
    fmodel = PyTorchModel(model, bounds=(0, 1), preprocessing=preprocessing)

    # get data and test the model
    # wrapping the tensors with ep.astensors is optional, but it allows
    # us to work with EagerPy tensors in the following
    images, labels = ep.astensors(*samples(fmodel, dataset="imagenet", batchsize=16))
    print("accuracy")
    print(accuracy(fmodel, images, labels))
    print("")

    attacks = [
        # fa.FGSM(),
        # fa.LinfPGD(),
        # fa.LinfBasicIterativeAttack(),
        # fa.LinfAdditiveUniformNoiseAttack(),
        # fa.L2CarliniWagnerAttack()
        fa.LinfDeepFoolAttack(),
    ]

    epsilons = [
        0.0,
        0.0005,
        0.001,
        0.0015,
        0.002,
        0.003,
        0.005,
        0.01,
        0.02,
        0.03,
        0.1,
        0.3,
        0.5,
        1.0,
    ]
    print("epsilons")
    print(epsilons)
    print("")

    attack_success = np.zeros((len(attacks), len(epsilons), len(images)), dtype=np.bool)
    for i, attack in enumerate(attacks):
        advs, _, success = attack(fmodel, images, labels, epsilons=epsilons)
        assert success.shape == (len(epsilons), len(images))
        success_ = success.numpy()
        assert success_.dtype == np.bool
        attack_success[i] = success_
        print(attack)
        print("  ", 1.0 - success_.mean(axis=-1).round(2))

        for eps, advs_ in zip(epsilons, advs):
            print(eps, accuracy(fmodel, advs_, labels))
            print((advs_ - images).norms.linf(axis=(1, 2, 3)).numpy())

    robust_accuracy = 1.0 - attack_success.max(axis=0).mean(axis=-1)
    print("")
    print("-" * 79)
    print("")
    print("worst case (best attack per-sample)")
    print("  ", robust_accuracy.round(2))
```

The output is like this 
![image](https://user-images.githubusercontent.com/16575311/84627242-1d5e8b80-af19-11ea-97c6-69632483399a.png)

The accuracies returned by function "attack" are in the green box while those calculated by adversarial images are in the blue box.  

This phenomenon also exists when using CWL2  (L2CarliniWagnerAttack()).
![image](https://user-images.githubusercontent.com/16575311/84627820-2ac84580-af1a-11ea-9522-5b3f3d18c928.png)

Since I am not familiar with this library, please correct me if I make any mistakes.

Thank you very much ~
